qiskit/transpiler/passes/consolidate_blocks.py

```diff
@@ -64,17 +65,42 @@ def run(self, dag):
             global_index_map[wire] = global_qregs.index(wire.register) + wire.index
 
         blocks = self.property_set['block_list']
-        nodes_seen = set()
+        # just to make checking if a node is in any block easier
+        all_block_nodes = [nd for bl in blocks for nd in bl]
```

Comment:
This could be made a `set`, to make membership tests a little more efficient. Does Collect2qBlocks guarantee that a node will only appear in at most one block?

Reply:
Yes, will do. It does as it keeps track of the nodes it has seen.